### PR TITLE
Adding Custom Fields Support

### PR DESCRIPTION
**Pull request by hermeswaldemarin**

### CODE REVIEW

File: sdk/Context.go [hunk 1]
```diff
@@ -422,6 +431,70 @@ func (c *Context) GetVariableKeys() (map[string]string, error) {
...
+
+func (c *Context) GetCustomFieldValueKeys() ([]string, error) {
+	var err = c.CheckReady(true)
+	if err != nil {
+		return nil, err
+	}
+
+	var keys []string
+
+	for _, experiment := range c.Data_.Experiments {
```
Comment (marcio-absmartly) [inline sdk/Context.go:457]: Shouldn't this be acquiring the data lock?

File: sdk/Context.go [hunk 2]
```diff
@@ -422,6 +431,70 @@ func (c *Context) GetVariableKeys() (map[string]string, error) {
...
+	return removeDuplicateValues(keys), nil
+}
+
+func (c *Context) GetCustomFieldValue(experimentName string, key string) interface{} {
+	var err = c.CheckReady(true)
+	if err != nil {
+		return nil
+	}
+
+	var customFieldValues = c.ContextCustomFields_[experimentName]
```
Comment (marcio-absmartly) [inline sdk/Context.go:478]: Same here?

File: sdk/Context.go [hunk 3]
```diff
@@ -422,6 +431,70 @@ func (c *Context) GetVariableKeys() (map[string]string, error) {
...
+	}
+	return nil
+}
+
+func (c *Context) GetCustomFieldValueType(experimentName string, key string) string {
+	var customFieldValues = c.ContextCustomFields_[experimentName]
+
+	var fieldType string
+	if customFieldValues != nil {
+		field, ok := customFieldValues[key]
```
Comment (marcio-absmartly) [inline sdk/Context.go:499]: And here?

**Review by marcio-absmartly — APPROVED** [review]

**Review by marcio-absmartly — CHANGES_REQUESTED** [review]
See comments

File: sdk/Context.go (hunk 1 above)
Comment (hermeswaldemarin) [reply sdk/Context.go:457]: Done

File: sdk/Context.go (hunk 2 above)
Comment (hermeswaldemarin) [reply sdk/Context.go:478]: Done

File: sdk/Context.go (hunk 3 above)
Comment (hermeswaldemarin) [reply sdk/Context.go:499]: Done

**Review by bmsilva — APPROVED** [review]

**Comment by hermeswaldemarin** [comment]
@marcio-absmartly  I need your review here since you commented.

**Review by marcio-absmartly — APPROVED** [review]

**Pull request merged**